backend/src/backend/primary/user_session_proxy.py

```diff
@@ -0,0 +1,123 @@
+import os
+import asyncio
+from typing import Dict, Any
+
+import httpx
+from starlette.requests import Request
+from starlette.responses import StreamingResponse
+from starlette.background import BackgroundTask
+
+from src.services.utils.authenticated_user import AuthenticatedUser
+
+LOCALHOST_DEVELOPMENT = os.environ.get("UVICORN_RELOAD") == "true"
+
+
+class RadixJobScheduler:
+    """Utility class to help with spawning Radix jobs on demand,
+    and provide correct URL to communicate with running Radix jobs"""
+
+    def __init__(self, name: str, port: int) -> None:
+        self._name = name
+        self._port = port
+
+        # TODO: This should be moved to Redis
+        # key: user_id, value: name of Radix job instance
+        self._existing_job_names: Dict[str, str] = {}
+
+    async def _active_running_job(self, user_id: str) -> bool:
+        """Returns true if there already is a running job for logged in user."""
+
+        existing_job_name = self._existing_job_names.get(user_id)
+        if not existing_job_name:
+            return False
+        elif LOCALHOST_DEVELOPMENT:
+            return True
+
+        async with httpx.AsyncClient() as client:
+            r = await client.get(f"http://{self._name}:{self._port}/api/v1/jobs/{existing_job_name}")
+
+        job = r.json()
+
+        if job.get("status") != "Running" or not job.get("started"):
+            return False
+
+        try:
+            httpx.get(f"http://{existing_job_name}:{self._port}/")
+        except (ConnectionRefusedError, httpx.ConnectError):
```

Comment:
```suggestion
        except (ConnectionRefusedError, httpx.ConnectError, httpx.ConnectTimeout):
```
We also observe a timeout sometimes on this call (when spinning up the container). Add this exception type to the group.